c++/getPos.cpp

```diff
@@ -0,0 +1,58 @@
+#include "Nbody.cpp"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+using namespace std;
+
+inline void simulate_n_bodies(Nbody bodies[], int num_bodies, int num_cycles) {
+    double x_val;
+    double y_val;
+    double z_val;
+    double starting_cycle = 2700;
+    std::cout << "Simulating Positions for " << num_cycles << std::endl;
+    for (int cycle = starting_cycle; cycle < num_cycles; cycle++) {
+        if (cycle % 10 == 0) {
+            std::cout << "Starting New Cycle " << cycle << std::endl;
+        }
+
+        std::stringstream ss;
+        ss << "c_positions/Nbodypositions_" << cycle << ".txt";
+        std::ofstream outfile(ss.str());
+        for (size_t i = 0; i < num_bodies; i++) {
+            std::vector<Nbody> relevantBodies;
+            for (size_t j = 0; j < num_bodies; j++) {
+                if (i != j) {
+                    relevantBodies.push_back(bodies[j]);
+                }
+            }
+            bodies[i].position(&relevantBodies[0], relevantBodies.size());
```

Comment:
You could at least pre-allocate the memory with something like `vector<Nbody> content(20000);`

Reply:
I was under the impression I had already done that somewhere in the project by doing cycles * bodies, will look into that.

Reply:
Well not in this location. Its still important that somewhere like here you do infact do that.